src/main/resources/com/google/api/codegen/nodejs/index.snip

```diff
@@ -0,0 +1,43 @@
+@extends "nodejs/common.snip"
+
+@snippet generate(index)
+  /*
+   {@copyright()}
+   */
+  'use strict';
+
+  @join require : index.requireViews
+    var {@require.clientName} = require('./{@require.fileName}');
+  @end
+  var gax = require('google-gax');
+  var extend = require('extend');
+  @if index.hasMultipleServices
+    var union = require('lodash.union');
+  @end
+
+  function {@index.apiVersion}(options) {
+    options = extend({
+      scopes: {@index.apiVersion}.ALL_SCOPES
+    }, options);
+    var gaxGrpc = gax.grpc(options);
+    var result = {};
+    @join require : index.requireViews
+      extend(result, {@require.clientName}(gaxGrpc));
+    @end
+    return result;
+  }
+
+  {@index.apiVersion}.SERVICE_ADDRESS = {@index.primaryService.clientName}.SERVICE_ADDRESS;
+  @if index.hasMultipleServices
+    {@index.apiVersion}.ALL_SCOPES = union(
+      @join require : index.requireViews on ", ".add(BREAK)
+        {@require.clientName}.ALL_SCOPES
+      @end
+    );
+  @else
+    {@index.apiVersion}.ALL_SCOPES = {@index.primaryService.clientName}.ALL_SCOPES;
```

Comment:
This change would be out of the scope of this PR so I prefer to do it in a separate one. But I suspect it would have a negative impact on the readability of the code. Imagine we have multiple interfaces and we group all scopes in this index file, the scope list could be very long and therefore not that readable.